src/io/tests/ExportVTUTest.cpp

```diff
@@ -19,27 +19,32 @@ class Vertex;
 } // namespace mesh
 } // namespace precice
 
-// void ExportVTKXMLTest:: run()
-// {
-//   PRECICE_TRACE();
-//   typedef utils::Parallel Par;
-//   if (Par::getCommunicatorSize() > 3){
-//     const std::vector<int> ranksWanted = {0, 1, 2, 3};
-//     MPI_Comm comm = Par::getRestrictedCommunicator(ranksWanted);
-//     if (Par::getProcessRank() <= 3){
-//       Par::setGlobalCommunicator(comm);
-//       testMethod(testExportPolygonalMesh);
-//       testMethod(testExportTriangulatedMesh);
-//       Par::setGlobalCommunicator(Par::getCommunicatorWorld());
-//     }
-//   }
-// }
-
 BOOST_AUTO_TEST_SUITE(IOTests)
 
 using namespace precice;
 
-BOOST_AUTO_TEST_SUITE(VTKXMLExport)
+BOOST_AUTO_TEST_SUITE(VTUExport)
+
+BOOST_AUTO_TEST_CASE(ExportPolygonalMeshSerial)
+{
+  PRECICE_TEST(""_on(1_rank).setupMasterSlaves());
+  int             dim = 2;
+  mesh::Mesh      mesh("MyMesh", dim, testing::nextMeshID());
+  mesh::Vertex &  v1      = mesh.createVertex(Eigen::VectorXd::Constant(dim, 0.0));
+  mesh::Vertex &  v2      = mesh.createVertex(Eigen::VectorXd::Constant(dim, 1.0));
+  Eigen::VectorXd coords3 = Eigen::VectorXd::Constant(dim, 0.0);
+  coords3(0)              = 1.0;
+  mesh::Vertex &v3        = mesh.createVertex(coords3);
```

Comment:
I understand that here you are constructing:
- v1 at (0.0, 0.0)
- v2 at (1.0, 1.0)
- v3 at (1.0, 0.0)

which, for `dim = 3`, would become:
- v1 at (0.0, 0.0, 0.0)
- v2 at (1.0, 1.0, 1.0)
- v3 at (1.0, 0.0, 0.0)

This confused me a bit, especially due to the name `coords3`, which is not really used further down (could be named, e.g. `v3_coords`). I assume it is not possible / good practice to move a `mesh::Vertex` after creating it?

Also, is line 37 properly formatted? Strange that the spacing is different.

Reply:
I have to admit that I simply copied this testcase from the `ExportVTK` tests. I'll look into them.